### PR TITLE
[PLA-9401] Configure the default design space.

### PR DESCRIPTION
**Pull request by anoto-moniz**
Allow the user to provide configuration flags when creating a default
design space. This also switches the method to use v2/design-space/default.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [X] I have added tests for 100% coverage
- [X] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [X] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/design_space.py [hunk 1]
```diff
@@ -64,7 +64,11 @@ def update(self, model: DesignSpace) -> DesignSpace:
         self._validate_write_request(model)
         return AbstractModuleCollection.update(self, model)
 
-    def create_default(self, *, predictor_id: UUID) -> DesignSpace:
+    def create_default(self, *,
```
Comment (kroenlein) [inline src/citrine/resources/design_space.py:67]:
```suggestion
    def create_default(self, 
                       *,
```
PEP has all args including the * on a new line.

File: tests/resources/test_design_space.py [hunk 2]
```diff
@@ -116,13 +117,76 @@ def test_create_default(valid_product_design_space_data,
...
+
+    assert session.num_calls == 1
+    assert session.last_call == expected_call
+    
     assert default_design_space.dump() == valid_product_design_space.dump()
 
+
+@pytest.mark.parametrize("config_flags", itertools.product((True, False), repeat=3))
+def test_create_default_with_config(valid_product_design_space_data, valid_product_design_space, config_flags):
+    ingredient_fractions, label_fractions, label_count = config_flags
```
Comment (kroenlein) [inline tests/resources/test_design_space.py:149]: Wouldn't make more sense to:
```suggestion
@pytest.mark.parametrize("ingredient_fractions", (True, False))
@pytest.mark.parametrize("label_fractions", (True, False))
@pytest.mark.parametrize("label_count", (True, False))
def test_create_default_with_config(valid_product_design_space_data, valid_product_design_space, ingredient_fractions, label_fractions, label_count):
```

File: tests/resources/test_design_space.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
-from copy import deepcopy
+import itertools
```
Comment (kroenlein) [inline tests/resources/test_design_space.py:1]:
```suggestion
```
if you take the parameterization suggestion.

**Review by kroenlein — COMMENTED** [review]
Formatting suggestions.

File: tests/resources/test_design_space.py (hunk 2 above)
Comment (anoto-moniz) [reply tests/resources/test_design_space.py:149]: *shrugs* Sure. They're the exact same, so I don't really care. As I'm making the tweak to the `*` formatting above, I'll make this change, too.

**Review by kroenlein — APPROVED** [review]
LGTM

**Pull request merged**